app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Eav/Source.php

```diff
@@ -40,6 +53,8 @@ public function __construct(
         \Magento\Eav\Model\Config $eavConfig,
         \Magento\Framework\Event\ManagerInterface $eventManager,
         \Magento\Catalog\Model\ResourceModel\Helper $resourceHelper,
+        \Magento\Eav\Api\AttributeRepositoryInterface $attributeRepository,
```

Comment:
I understand the reason for this ask, but I was under the impression that the direct use of ObjectManager was discouraged as per the rules in the documentation [here](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/object-manager.html#usage-rules). And the warning indicates any usage in core classes is due to backwards compatibility and should be updated eventually. Is this an exception, or is there a particular reason why we should use it here?

Reply:
That is an exception metioned in the referenced guide: "You can use the ObjectManager to maintain backward compatibility for a constructor." We should use it here because there is no other way to ensure the properties are assigned and the implementation is backward compatible.

Reply:
Ok that makes sense, thanks for the explanation.